cmake/thirdparty/SetupHIP.cmake

```diff
@@ -15,24 +15,76 @@ find_package(HIP REQUIRED)
 message(STATUS "HIP version:      ${HIP_VERSION_STRING}")
 message(STATUS "HIP platform:     ${HIP_PLATFORM}")
 
-if("${HIP_PLATFORM}" STREQUAL "hcc" OR "${HIP_PLATFORM}" STREQUAL "amd")
-	set(HIP_RUNTIME_DEFINE "__HIP_PLATFORM_HCC__")
-elseif("${HIP_PLATFORM}" STREQUAL "nvcc")
-	set(HIP_RUNTIME_DEFINE "__HIP_PLATFORM_NVCC__")
+set(HIP_RUNTIME_INCLUDE_DIRS "${HIP_ROOT_DIR}/include")
+if(${HIP_PLATFORM} STREQUAL "hcc")
+	set(HIP_RUNTIME_DEFINES "-D__HIP_PLATFORM_HCC__")
+    set(HIP_RUNTIME_LIBRARIES "${HIP_ROOT_DIR}/lib/libhip_hcc.so")
+elseif(${HIP_PLATFORM} STREQUAL "clang" OR ${HIP_PLATFORM} STREQUAL "amd")
+    set(HIP_RUNTIME_DEFINES "-D__HIP_PLATFORM_HCC__;-D__HIP_ROCclr__")
+    set(HIP_RUNTIME_LIBRARIES "${HIP_ROOT_DIR}/lib/libamdhip64.so")
+elseif(${HIP_PLATFORM} STREQUAL "nvcc")
+    set(HIP_RUNTIME_DEFINES "-D__HIP_PLATFORM_NVCC__")
+    find_package(cudatoolkit)
+    set(HIP_RUNTIME_LIBRARIES "${CUDAToolkit_LIBRARY_DIR}/libcudart.so")
+    set(HIP_RUNTIME_INCLUDE_DIRS "${HIP_RUNTIME_INCLUDE_DIRS};${CUDAToolkit_INCLUDE_DIR}")
 endif()
 if ( IS_DIRECTORY "${HIP_ROOT_DIR}/hcc/include" ) # this path only exists on older rocm installs
-        set(HIP_RUNTIME_INCLUDE_DIRS "${HIP_ROOT_DIR}/include;${HIP_ROOT_DIR}/hcc/include" CACHE STRING "")
+    set(HIP_RUNTIME_INCLUDE_DIRS "${HIP_ROOT_DIR}/include;${HIP_ROOT_DIR}/hcc/include" CACHE STRING "")
 else()
-        set(HIP_RUNTIME_INCLUDE_DIRS "${HIP_ROOT_DIR}/include" CACHE STRING "")
+    set(HIP_RUNTIME_INCLUDE_DIRS "${HIP_ROOT_DIR}/include" CACHE STRING "")
 endif()
-set(HIP_RUNTIME_COMPILE_FLAGS "${HIP_RUNTIME_COMPILE_FLAGS};-D${HIP_RUNTIME_DEFINE};-Wno-unused-parameter")
+set(HIP_RUNTIME_COMPILE_FLAGS "${HIP_RUNTIME_COMPILE_FLAGS};-Wno-unused-parameter")
+
+set(_hip_compile_flags " ")
+if (ENABLE_CLANG_HIP)
+    if (NOT (${HIP_PLATFORM} STREQUAL "clang"))
+        message(FATAL_ERROR "ENABLE_CLANG_HIP requires HIP_PLATFORM=clang")
+    endif()
+    set(_hip_compile_flags -x;hip)
+    # Using clang HIP, we need to construct a few CPP defines and compiler flags
+    foreach(arch ${BLT_CLANG_HIP_ARCH})
+        string(TOUPPER ${arch} UPARCH)
+        string(TOLOWER ${arch} lowarch)
```

Comment:
Minor: should these variable names be prefixed with an underscore?

Reply:
Added underscores to all the internal variables in this look. It's a trivial change, and it doesn't hurt to be a bit more clear.